.github/workflows/ci.yml

```diff
@@ -115,6 +121,10 @@ jobs:
           extensions: ${{ env.PHP_EXTENSIONS }}
           ini-values: ${{ matrix.php-ini-values }}
 
+      - name: Update composer to snapshot
+        if: matrix.php-version == '8.4'
+        run: composer self-update --snapshot
```

Comment:
Until 2.7.8 is out 8.4 should use `snapshot` https://github.com/jsonrainbow/json-schema/pull/717#issuecomment-2212513129

Reply:
Somehow `tools: composer:snapshot` does not work

Reply:
Fixed by passing value from matrix